### ISSUE TITLE
Models always predict nearly identical values with only minor variations on financial data

### ISSUE THREAD

**Issue by dqgdqg** [thuml/Time-Series-Library#656]
Thank you for your open-source contributions to the community!

I'm currently working on applying codes to my own financial time series datasets (like Exchange), which are more high-frequency and involve larger data volumes (over Terabytes data). The task is a regression (only predict one value given a time series). I have trained several models, but they all seem to predict nearly identical values with only minor variations. The model might be minimizing the loss function (e.g., Mean Squared Error, MSE) by predicting the average value, as it reduces overall error.

Could you provide any suggestions or strategies to address this issue? Thanks in advance!

**Comment by clivinn-shla81092** [comment]
Hello, we tried to solve the issue.

This is what we did:

Update the Exp_Long_Term_Forecast class in exp/exp_long_term_forecasting.py to include data preprocessing for high-frequency data, add a custom loss function, and implement gradient clipping.

You can review changes in this commit: https://github.com/clivinn-shla81092/thuml-Time-Series-Library-656/commit/84e3444cfbe7cc41ebac4339bf0303612371b812.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.
